### PR TITLE
[SPARK-40070][BUILD] Upgrade Apache `ZooKeeper` to 3.8.0

### PR DESCRIPTION
**Pull request by bjornjorgensen**
### What changes were proposed in this pull request?
Upgrade Apache ZooKeeper from 3.6.2 to 3.8.0

### Why are the changes needed?
SNYK want to Upgrade org.apache.zookeeper:zookeeper from 3.6.2 to 3.8.0

The recommended version is 4 versions ahead of your current version.
The recommended version was released 5 months ago, on 2022-02-25.

[Releases notes.](https://zookeeper.apache.org/releases.html) 

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?
Pass GA.

### CODE REVIEW

File: dev/deps/spark-deps-hadoop-2-hive-2.3 [hunk 1]
```diff
@@ -190,6 +190,8 @@ log4j-1.2-api/2.18.0//log4j-1.2-api-2.18.0.jar
 log4j-api/2.18.0//log4j-api-2.18.0.jar
 log4j-core/2.18.0//log4j-core-2.18.0.jar
 log4j-slf4j-impl/2.18.0//log4j-slf4j-impl-2.18.0.jar
+logback-classic/1.2.10//logback-classic-1.2.10.jar
```
Comment (LuciferYang) [inline dev/deps/spark-deps-hadoop-2-hive-2.3:193]: Won't this conflict with log4j2?

Comment (LuciferYang) [reply dev/deps/spark-deps-hadoop-2-hive-2.3:193]: also cc @viirya

File: dev/deps/spark-deps-hadoop-2-hive-2.3 [hunk 2]
```diff
@@ -206,6 +208,7 @@ netty-common/4.1.77.Final//netty-common-4.1.77.Final.jar
 netty-handler/4.1.77.Final//netty-handler-4.1.77.Final.jar
 netty-resolver/4.1.77.Final//netty-resolver-4.1.77.Final.jar
 netty-tcnative-classes/2.0.52.Final//netty-tcnative-classes-2.0.52.Final.jar
```
Comment (LuciferYang) [inline dev/deps/spark-deps-hadoop-2-hive-2.3:210]: What is the difference between `netty-tcnative-classes` and `netty-tcnative`? netty 4.1.77 requires `netty-tcnative-classes 2.0.52.Final`

**Comment by LuciferYang** [comment]
@bjornjorgensen Could you do some manually test using Maven with `hadoop-2` profile ?

**Comment by bjornjorgensen** [comment]
@LuciferYang Thank you. I have just run `./dev/test-dependencies.sh --replace-manifest` and the tests have passed. 
hm.. there are more than you that ask for the same thing. https://github.com/apache/zookeeper/pull/1867 and https://github.com/apache/zookeeper/pull/1873

"_ZooKeeper clients from 3.5.x onwards are fully compatible with 3.8.x servers. The upgrade from 3.6.x and 3.7.x can be executed as usual, no particular additional upgrade procedure is needed. ZooKeeper 3.8.x clients are compatible with 3.5.x, 3.6.x and 3.7.x servers as long as you are not using new APIs not present these versions._"
*present in these

But should we upgrade to 3.6.3? or 3.7 or?

**Pull request closed without merging**